## Task 4  — Country Traffic & Customer Analysis

### Business Question

Which countries have the highest number of customers and the highest total download traffic?

### Analysis

Group customers by country, calculate the number of customers, average download traffic, and total download traffic for each country. Sort the results by total download traffic in descending order and select the top 5 countries.

In [8]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
customers.groupby("country")["unique_id"].count()
customers.groupby("country")["gb_downloaded"].mean()
customers.groupby("country").agg({
    "unique_id": "count",
    "gb_downloaded": ["mean", "sum"]
}).sort_values(
    ("gb_downloaded","sum"),
    ascending=False
).head(5)

unique_id gb_downloaded            
                   count          mean         sum
country                                           
Ukraine             1032   1562.133547  1612121.82
Germany             1027   1534.529747  1575962.05
Australia           1030   1517.334117  1562854.14
Spain               1019   1526.638440  1555644.57
United Kingdom       977   1573.757963  1537561.53

### Result

Ukraine has the highest total download traffic among all countries at 1,612,121.82 GB, followed by Germany (1,575,962.05 GB) and Australia (1,562,854.14 GB). The top 5 countries have between 977 and 1,032 customers each.

### Business Insight

The top 5 countries have similar customer counts and average download traffic, so differences in total traffic are mainly driven by the number of customers and their usage levels.